straxen/plugins/acqmon_processing.py

```diff
@@ -90,44 +94,122 @@ def setup(self):
 
     def get_window_size(self):
         # Give a very wide window
-        return (self.config['max_veto_gap'] * 100)
+        return (self.config['max_veto_window'] * 100)
 
     def compute(self, aqmon_hits):
         hits = aqmon_hits
 
-        # Make a dict to populate later
-        res_temp = dict()
-
-        for i, veto in enumerate(self.veto_names):
-            veto_hits = channel_select(hits, self.channel_map[veto + 'stop'], self.channel_map[veto + 'start'])
-            vetos = self.merge_vetos(veto_hits, gap=self.config['min_veto_gap'], \
-                                     dtype=strax.time_fields, t=0)
-
-            # Check that we found a veto interval and update the resulting dict
-            if len(vetos):
-                res_temp.setdefault("time", []).extend(vetos['time'])
-                res_temp.setdefault("endtime", []).extend(vetos['endtime'])
-                res_temp.setdefault("veto_interval", []).extend((vetos['endtime'] - vetos['time']))
-                res_temp.setdefault("veto_type", []).extend([veto + 'veto'] * len(vetos))
-
-        res = strax.dict_to_rec(res_temp, dtype=self.dtype)
-        res.sort(order='time')
-
-        return res
-
-    @staticmethod
-    def merge_vetos(channels, gap, dtype, t):
-        if len(channels):
-            start, stop = strax.find_peak_groups(channels, gap_threshold=gap)
-            result = np.zeros(len(start), dtype=dtype)
-            result['time'] = start
-            result['endtime'] = stop
-        else:
-            result = np.zeros(0, dtype=dtype)
+        res = defaultdict(list)
+        
+        for n, name in enumerate(self.veto_names):
+            veto_hits_start = channel_select_(hits, self.channel_map[name + 'start'])
+            veto_hits_stop = channel_select_(hits, self.channel_map[name + 'stop'])
+            
+            # Here we rely on the fact that for each start, there is a single stop that 
+            # follows it in time. If this is not true, our hardware does not work. 
+            if len(veto_hits_start):
+                for t, time in enumerate(veto_hits_start['time']): 
+                    # Find the time of stop_j that is closest to time of start_i
+                    inx = np.searchsorted(veto_hits_stop['time'], time, side = 'right') 
```

Comment:
If I follow your comment from above I would expect that I get for each index `t` (which is not used btw) the correct start and stop. So why do we have to sort here?